### PR TITLE
Add read-only vector GTFS (General Transit Feed Specification) driver

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: ogr/ogrsf_frmts/CMakeLists.txt [hunk 1]
```diff
@@ -96,6 +96,8 @@ ogr_dependent_driver(cad "OpenCAD" "GDAL_USE_OPENCAD OR GDAL_USE_OPENCAD_INTERNA
 ogr_dependent_driver(parquet "Parquet" "GDAL_USE_PARQUET")
 ogr_dependent_driver(arrow "Arrow" "GDAL_USE_ARROW")
 
+ogr_dependent_driver(gtfs "GTFS" "OGR_ENABLE_DRIVER_CSV")
```
Comment (elpaso) [inline ogr/ogrsf_frmts/CMakeLists.txt:99]: Shouldn't it be OGR_ENABLE_DRIVER_GTFS ?

File: ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp [hunk 2]
```diff
@@ -0,0 +1,724 @@
...
+    int m_nLatIdx = -1;
+    int m_nLonIdx = -1;
+    bool m_bIsTrips = false;
+    bool m_bPrepared = false;
+    std::map<std::string, std::pair<double, double>> m_oMapStopIdToLonLat{};
+    std::map<std::string, std::map<int, std::string>> m_oMapTripIdToStopIds{};
+
+    void PrepareTripsData();
+
+  public:
```
Comment (elpaso) [inline ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp:88]: I'm not sure about GDAL coding conventions, but I would expect to find the `public` interface first and an explicit `private` secion below it.

File: ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp [hunk 3]
```diff
@@ -0,0 +1,724 @@
...
+                poFeature->GetFieldAsInteger(nStopSequenceIdx);
+            if (pszStopId && pszTripId)
+            {
+                m_oMapTripIdToStopIds[pszTripId][nStopSequence] = pszStopId;
+            }
+        }
+    }
+    catch (const std::exception &)
+    {
+        CPLError(CE_Failure, CPLE_OutOfMemory, "Not enough memory");
```
Comment (elpaso) [inline ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp:264]: Just wondering: is OOM the only case when the `try` above can `throw` ?

File: ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp [hunk 4]
```diff
@@ -0,0 +1,724 @@
...
+        poFeature->SetFID(poSrcFeature->GetFID());
+        auto poSrcLayerDefn = m_poUnderlyingLayer->GetLayerDefn();
+        for (int i = 0; i < nFieldCount; ++i)
+        {
+            const auto eType = m_poFeatureDefn->GetFieldDefn(i)->GetType();
+            if (poSrcLayerDefn->GetFieldDefn(i)->GetType() == eType)
+            {
+                poFeature->SetField(i, poSrcFeature->GetRawFieldRef(i));
+            }
+            else if (eType == OFTDate)
```
Comment (elpaso) [inline ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp:295]: Maybe a switch? I see only date and int are actually handled and default to string.

**Review by elpaso — APPROVED** [review]

File: ogr/ogrsf_frmts/CMakeLists.txt (hunk 1 above)
Comment (rouault) [reply ogr/ogrsf_frmts/CMakeLists.txt:99]: no, the GTFS driver requires (at runtime) the CSV driver

File: ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp (hunk 2 above)
Comment (rouault) [reply ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp:88]: we don't have that much conventions regarding that. The pattern with private first is actually quite common in GDAL sources

File: ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp (hunk 3 above)
Comment (rouault) [reply ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp:264]: I think so. The risk is here is too many objects beeing put in the std::map. I'll change it to specifically catch std::bad_alloc though

File: ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp (hunk 4 above)
Comment (rouault) [reply ogr/ogrsf_frmts/gtfs/ogrgtfsdriver.cpp:295]: A switch construct isn't the most natural one here given that the first ``if`` doesn't test eType against a constant but against an expression

**Pull request merged**